components/rules-watcher/kustomization.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+
+patches:
+  - path: thanos-rule.yaml
+
+configMapGenerator:
+  - name: rules-watcher-config
+    files:
+      - rules-watcher.sh
```

Comment:
I like this style guide for shell scripts, it has a section on script naming: https://google.github.io/styleguide/shellguide.html#file-extensions